luigi/worker.py

```diff
@@ -197,7 +197,13 @@ def run(self):
             else:
                 with self._forward_attributes():
                     new_deps = self._run_get_new_deps()
-                status = DONE if not new_deps else PENDING
+                if not new_deps:
+                    if not self.check_complete_on_run or self.task.complete():
+                        status = DONE
+                    else:
+                        raise TaskException("Task finished running, but complete() is still returning false.")
```

Comment:
Do we want to raise an exception in this case? Or saying the task `FAILED`?

Or will the task already be FAILED? And we just want to raise an exception for the reason why?

Reply:
Initially I had it just marked as 'failed', but it seemed that raising an exception was the standard way to do this (given that the default `on_failure` callback doesn't do anything with its arguments and just processes the currently-being-handled exception). I can switch back to manually marking the failure if that's what's preferred.